### DISCUSSION TITLE
Missing Argument with Prisma and Apollo Graphql

### DISCUSSION THREAD

**Discussion by Esorethewarrior0623** [prisma/prisma#11422] [Q&A] [answered]
I am trying to do a simple mutation from Apollo Graphql into my Postgres database via prisma. 

This is my code in the resolver: 
```ts
const {v4: uuid} = require('uuid');
import { PrismaClient } from "@prisma/client"
const prisma = new PrismaClient();

 createNewSampleProduct:(parent:any, args:any) => {
        return prisma.sampleProduct.create({
            data: {
                id:uuid(),
                name: args.name,
            }
        })
    }
```

Very simple, however when I run 
![image](https://user-images.githubusercontent.com/79986478/151234423-9b0d0df4-822b-438a-8580-0646dfbcb316.png)

I get a create invocation Error and an Argument Error:
![image](https://user-images.githubusercontent.com/79986478/151234598-8076ab6c-45b9-4915-b375-64713e77a2ee.png)

I am just want to learn how to use prisma via apollo graphql as a means of inserting data into my database to make sure it works. Any help would be great!

**Comment by austincrim** [comment, marked as answer]
Hey @Esorethewarrior0623 👋 ,

Are you certain the `name` argument is coming through your query correctly? The error message suggests that the value isn't getting passed to your query. Could you try logging `args.name`?

**Reply by Esorethewarrior0623** [reply]
When I console it, it comes back as undefined when I run the Apollo server Action

**Reply by austincrim** [reply]
Could you try logging `args.input.name`? It looks like you are actually passing an `input` object that contains a `name` property. 

Otherwise, I would encourage you to look through the [Apollo docs](https://www.apollographql.com/docs/) as it seems like this is an issue with the GraphQL setup.

Thanks!

**Reply by Esorethewarrior0623** [reply]
that worked! Thank you so much!

**Reply by rahXephonz** [reply]
Im sure you are using typescript, recommend to create a interface or type in every args or context it Will make help you to writing code.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
